oak-search-elastic/src/test/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticConnectionRule.java

```diff
@@ -75,6 +80,46 @@ protected void after() {
         }
     }
 
+    public ElasticConnectionModel getElasticConnectionModel() {
+        return elasticConnectionModel;
+    }
+
+    private void initializeElasticConnectionModel(String elasticConnectionString) {
+        try {
+            URI uri = new URI(elasticConnectionString);
+            String host = uri.getHost();
+            String scheme = uri.getScheme();
+            int port = uri.getPort();
+            String query = uri.getQuery();
+
+            String apiKey = null;
+            String apiSecret = null;
+            if (query != null) {
+                apiKey = query.split(",")[0].split("=")[1];
```

Comment:
As far as I understand, this assumes that the first parameter is always the API key, and the second always the secret... Example: "key=abc,secret=def". But if someone manually changes that to "secret=def,key=abc"?

What about using https://stackoverflow.com/questions/14768171/convert-string-representing-key-value-pairs-to-map to convert the key-value pair to a map and then retrieve the key and secret from the map?

Reply:
Yes, this was already implemented and in comment we had defined url so that this logic will work.
`// Set this connection string as
    // <scheme>://<hostname>:<port>?key_id=<>,key_secret=<>
    // key_id and key_secret are optional in case the ES server
    // needs authentication
    // Do not set this if docker is running and you want to run the tests on docker instead.`

But yes we can change the logic to make it to be more generic.

Reply:
Done